src/main/java/io/codelex/flightplanner/service/FlightServiceImplementation.java

```diff
@@ -0,0 +1,107 @@
+import io.codelex.flightplanner.exception.DuplicateFlightException;
+import io.codelex.flightplanner.exception.SameAirportException;
+import io.codelex.flightplanner.exception.StrangeDatesException;
+import io.codelex.flightplanner.repository.FlightRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class FlightServiceImplementation implements FlightService {
+    private final FlightRepository flightRepository;
+
+    public FlightServiceImplementation(FlightRepository flightRepository) {
+        this.flightRepository = flightRepository;
+    }
+
+    public void add(Flight flight) throws DuplicateFlightException, SameAirportException, StrangeDatesException {
+        validateFlight(flight);
+        flightRepository.add(flight);
+    }
+
+    public Flight getFlightById(int id) {
+        return flightRepository.getFlightById(id);
+    }
+
+    public void deleteFlightById(int id) {
+        flightRepository.deleteFlightById(id);
+    }
+
+    public void clearFlightList() {
+        flightRepository.clearFlightList();
+    }
+
+    public SearchFlightResponse searchFlights(SearchFlightsRequest searchFlightsRequest) throws SameAirportException {
+        validateAirportRoute(searchFlightsRequest.getFrom(), searchFlightsRequest.getTo());
+        validateSearchFlightRequest(searchFlightsRequest);
+        List<Flight> searchFlightsResponseContent = flightRepository.getAll().stream()
+                .filter(flight -> flight.getFrom().getAirport().equals(searchFlightsRequest.getFrom()))
+                .filter(flight -> flight.getTo().getAirport().equals(searchFlightsRequest.getTo()))
+                .filter(flight -> flight.getDepartureTime().toLocalDate().equals(searchFlightsRequest.getDepartureDate()))
+                .toList();
+        return new SearchFlightResponse(searchFlightsResponseContent.size(), 0, searchFlightsResponseContent);
+    }
+
+    public List<Airport> getPartialFlightSearchByAirport(String search) {
+        String formattedSearch = search.toLowerCase().trim();
+        return flightRepository.getAll().stream()
+                .map(Flight::getFrom)
+                .filter(from -> from.getCountry().toLowerCase().contains(formattedSearch)
+                        || from.getCity().toLowerCase().contains(formattedSearch)
+                        || from.getAirport().toLowerCase().contains(formattedSearch))
+                .toList();
+    }
+
+    public List<Flight> getAll() {
+        return flightRepository.getAll();
+    }
+
+    private void validateFlight(Flight flight) throws DuplicateFlightException, SameAirportException, StrangeDatesException {
+        if (flight.getFrom() == null || flight.getTo() == null || flight.getCarrier() == null || flight.getCarrier().isEmpty() || flight.getDepartureTime() == null || flight.getArrivalTime() == null) {
+            throw new NullPointerException("Flight object cannot have null or empty fields.");
```

Comment:
Nav korekti mest šeit NullPointerException - jo tā kļūda ir ļoti konkrēta - viņa liecina to, ka ir tikusi izsaukta metode objektam, kurš ir null. Būtu jāmet kāda cita - var piemēram IllegalArgumentException.